### ISSUE TITLE
[iOS] Shadow issue when Transitioning

### ISSUE THREAD

**Issue by DovletAmanov** [software-mansion/react-native-reanimated#626]
Whenever I use shadow for transitioning view, shadow appears weirdly in iOS and in Android it seems transition is not working smoothly

RN: 0.61.5
react-native-reanimated: 1.7.0

Here is a snack: https://snack.expo.io/@dovletamanov/reanimated-transition-shadow-issue

Also you can try this code below with newly created non-expo project:

```
import React, {useState, useRef} from 'react';
import {ScrollView, View, Text, TouchableWithoutFeedback} from 'react-native';

import Animated, {Transition, Transitioning} from 'react-native-reanimated';

const App = () => {
  return (
    <ScrollView contentContainerStyle={{flex: 1}}>
      <View style={{flex: 1, alignItems: 'center', justifyContent: 'center'}}>
        {[0, 1, 2,3].map((item, index) => (
          <ListItem key={index} item={item} />
        ))}
      </View>
    </ScrollView>
  );
};

export default App;

const ListItem = ({item}) => {
  const [active, setActive] = useState(false);
  const ref = useRef(null);
  const transition = (
    <Transition.Change interpolation="easeInOut" durationMs={300} />
  );
  const shadowOffset = 6;
  return (
    <Transitioning.View
      {...{ref, transition}}>
      <TouchableWithoutFeedback
        onPress={() => {
          if (ref && ref.current) {
            ref.current.animateNextTransition();
            setActive(prev => !prev);
          }
        }}>
        <Animated.View  style={{
            marginVertical: 20,
            elevation: 2,
            shadowColor: 'black',
            shadowOffset: {width: 0, height: shadowOffset},
            shadowOpacity: 0.034 * shadowOffset,
            shadowRadius: 0.627 * shadowOffset,
            backgroundColor: 'white',
        }}>
          <View style={{height: 60, width: 300, backgroundColor: 'orange'}} />
          <Animated.View
            style={{height: active ? 400 : 0, backgroundColor: 'tomato'}}>
            <Text>{item}</Text>
          </Animated.View>
        </Animated.View>
      </TouchableWithoutFeedback>
    </Transitioning.View>
  );
};

```

**Comment by antstu** [comment]
I'm having the same issue - did you find a solution for this?

**Comment by dominguezcm** [comment]
Try out your Animated View inside another View classic and put the shadow style to this View.

**Comment by jakub-gonet** [comment]
We discussed Transitions internally in our team. Considering that v2 makes creating animations easier and Transitions was an experimental API for a long time, we decided to leave Transitions as they are.
 
We won’t actively fix bugs in them, but we’ll happily merge any PR with fixes. We’re also thinking about rewriting Transitions in v2 sometime in the future.

I’m closing this issue, but if you have further questions, I will be happy to help.
